### ISSUE TITLE
Tweening arrays and objects?

### ISSUE THREAD

**Issue by gautamadude** [tweenjs/tween.js#78]
So I have these "from" and "to" objects that I need to tween:

```javascript
var from = {x: 0, y: {a: 0, b: 0}, z:[0, 0], color: "#000000"};
var to = {x: 10, y: {a: 20, b: 30}, z:[40, 50], color: "#ffffff"};
```

Tween.js doesn't handle the "y", "z" or "color" values out of the box, but would it make sense to do so? Maybe not colors, but what about objects and arrays? And perhaps even nested objects and arrays like so:

```javascript
{x: 0, y: {a: {t: 0, k:0}, b: 0}, z:[[5,2], [10]], color: "#000000"};
```

So far to deal with this I came up with the following in a hurry, and I still need to implement support for nested arrays and objects (also if you know of a better way I'd be interested):

http://jsfiddle.net/gautamadude/YRbQ3/
```javascript
var from = {x: 0, y: {a: 0, b: 0}, z:[0, 0], color: "#000000"};
var to = {x: 10, y: {a: 20, b: 30}, z:[40, 50], color: "#ffffff"};
var speed = 2000;
var easing = TWEEN.Easing.Linear.None;
var delay = 0;
tweenWithObjectArrayAndColorSupport(from, to, speed, easing, delay);

function tweenWithObjectArrayAndColorSupport(from, to, speed, easing, delay) {
	var outFromTo = tweenSeperate(from, to); //Seperate the colors, array objects, and objects
											 //from the "from" and "to" objects.

	//Tween the remainder
	new TWEEN.Tween( from )
		.to( to, speed )
		.easing( easing )
		.delay(delay)
		.start();

	//Create Tweens for the separated values
	for (type in outFromTo.from) {
		for (attr in outFromTo.from[type]) {
			var vals = {"attr": attr, "type": type};
			createTween(vals, outFromTo.from[type][attr], outFromTo.to[type][attr], speed, easing, delay);
		}
	}
}

function tweenSeperate(from, to) {
	var vals = {};
	var objs = [from, to];
	for (o in objs) {
		var outColors = {};
		var outObjects = {};
		var outArrays = {};
		for (a in objs[o]) {
			if (typeof objs[o][a] == "string") {
				if (/(^#[0-9A-F]{6}$)|(^#[0-9A-F]{3}$)/i.test(objs[o][a])) {
					var t = tinycolor(objs[o][a])
					outColors[a] = t.toHsv();
					delete objs[o][a];
				}
			}
			if (Object.prototype.toString.call(objs[o][a]) == "[object Object]") {
				outObjects[a] = objs[o][a];
				delete objs[o][a];
			} else if (Object.prototype.toString.call(objs[o][a]) == "[object Array]") {
				outArrays[a] = {};
				var array = objs[o][a];
				delete objs[o][a];
				for (val in array) {
					outArrays[a][val] = array[val];
				}
			}
		}
		if (o == "0") vals.from = {"colors": outColors, "objects": outObjects, "arrays": outArrays};
		else vals.to = {"colors": outColors, "objects": outObjects, "arrays": outArrays};
	}
	return vals;
}

function createTween(vals, f, t, speed, easing, delay) {
	new TWEEN.Tween( f )
		.to( t, speed )
		.easing( easing )
		.onUpdate( function() {
			if (vals.type == "objects" && from[vals.attr] == undefined) {
				from[vals.attr] = {};
			} else if (vals.type == "arrays" && from[vals.attr] == undefined) {
				from[vals.attr] = [];
			}

			if (vals.type == "colors") {
				var t = tinycolor(this)
				from[vals.attr] = t.toHexString();
			} else {
				for (a in this) {
					if (a != "attr" & a != "type") {
						from[vals.attr][a] = this[a];
					}
				}
			}
		})
		.onComplete( function () {
			console.log(from); //Everything Tweened correctly!
		})
		.delay(delay)
		.start();
}

function r(frame) {
	requestAnimationFrame( r );
	TWEEN.update();
}
r();
```

The reason I need this is to tween KineticJS, Two.js and Pixi.js shape attributes.

So basically, would it make sense to implement support for tweening array and object values (nested or not) in Tween.js?
